resource/reapi/bindings/go/src/pkg/types/types.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+/*****************************************************************************\
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, LICENSE)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\*****************************************************************************/
+
+/*
+#include "reapi_cli.h"
+
+typedef enum {
+	MATCH_UNKNOWN = 0
+	MATCH_ALLOCATE = 1
+	MATCH_ALLOCATE_ORELSE_RESERVE = 2
+	MATCH_ALLOCATE_W_SATISFIABILITY = 3
+	MATCH_SATISFIABILITY = 4
+} match_op_enum;
```

Comment:
This looks good, but I wonder if we should/could use the match_op_t enum directly here? Here's how its defined:

```
enum class match_op_t { MATCH_ALLOCATE,
                        MATCH_ALLOCATE_W_SATISFIABILITY,
                        MATCH_ALLOCATE_ORELSE_RESERVE,
                        MATCH_SATISFIABILITY };
```

Reply:
Yes we should, that is left for you to implement! I don't dare venture into C++ land, it's very dangerous because I suck at it and will be eaten alive. :)

Reply:
and note my comment about the need for a match unknown case. In go when you instantiate a variable (and don't properly add in the type) it will be 0. We just need to distinguish that from what you have currently, which would have MATCH_ALLOCATE as 0 (which could either indicate it was set to that, or not set at all, we can't tell the difference)!

Reply:
I can play around with how the enum is declared and see if I can add a default value for place 0. Hopefully that won't break anything. If it does, what do you think about the idea of just having match_allocate be the default choice?

Reply:
I think that would be a question for the larger group - I would say the default should be something that doesn't lead to a change in state, e.g., maybe just asking if a request can be satisfied.

Reply:
Looks like adding a default option does not seem to break any tests, so this should work 

```
enum class match_op_t { MATCH_UNKNOWN,
                        MATCH_ALLOCATE,
                        MATCH_ALLOCATE_W_SATISFIABILITY,
                        MATCH_ALLOCATE_ORELSE_RESERVE,
                        MATCH_SATISFIABILITY };
```